src/prefect/runner/storage.py

```diff
@@ -276,14 +279,204 @@ def to_pull_step(self) -> Dict:
         return pull_step
 
 
+class RemoteStorage:
+    """
+    Pulls the contents of a remote storage location to the local filesystem.
+
+    Parameters:
+        url: The URL of the remote storage location to pull from. Supports
+            `fsspec` URLs. Some protocols may require an additional `fsspec`
+            dependency to be installed. Refer to the
+            [`fsspec` docs](https://filesystem-spec.readthedocs.io/en/latest/api.html#other-known-implementations)
+            for more details.
+        pull_interval: The interval in seconds at which to pull contents from
+            remote storage to local storage. If None, remote storage will perform
+            a one-time sync.
+        **settings: Any additional settings to pass the `fsspec` filesystem class.
+
+    Examples:
+        Pull the contents of a remote storage location to the local filesystem:
+
+        ```python
+        from prefect.runner.storage import RemoteStorage
+
+        storage = RemoteStorage(url="s3://my-bucket/my-folder")
+
+        await storage.pull_code()
+        ```
+
+        Pull the contents of a remote storage location to the local filesystem
+        with additional settings:
+
+        ```python
+        from prefect.runner.storage import RemoteStorage
+        from prefect.blocks.system import Secret
+
+        storage = RemoteStorage(
+            url="s3://my-bucket/my-folder",
+            # Use Secret blocks to keep credentials out of your code
+            key=Secret.load("my-aws-access-key"),
+            secret=Secret.load("my-aws-secret-key"),
+        )
+
+        await storage.pull_code()
+        ```
+    """
+
+    def __init__(
+        self,
+        url: str,
+        pull_interval: Optional[int] = 60,
+        **settings: Any,
+    ):
```

Comment:
If they provide a `key` and `secret`, should we be checking if the blocks exist _if_ blocks are being based in?

Reply:
Er, actually, it seems like you load it as you pass it in so I think it would fail then. Maybe it would make sense to add a test with an invalid block to see how that looks

Reply:
We check if any provided blocks have been saved in `.to_pull_step`. Using `Secret` isn't strictly necessary unless you're using `.deploy` because we don't persist the settings serverside when using `RemoteStorage` with `.serve`.

Reply:
We resolve the block values with this function:
```python
def replace_blocks_with_values(obj: Any) -> Any:
    if isinstance(obj, Block):
        if hasattr(obj, "get"):
            return obj.get()
        if hasattr(obj, "value"):
            return obj.value
        else:
            return obj.dict()
    return obj
```
which doesn't require the block to be saved because were expecting a block instance. The user would need to call `.load` before passing in if they're loading blocks from the server.

Reply:
Added a test for unsaved blocks when calling `.to_pull_step` in https://github.com/PrefectHQ/prefect/pull/11072/commits/edaefd303714c8ee0db144dd0be96a50a6d6bfbf

Reply:
ah yeah I realized that now, thanks for the clarification though!